src/app/todos/page.js

```diff
@@ -36,9 +36,24 @@ export default function ToDos() {
     }
 
     function removeTodo({ index }) {
+        const todoToRemove = todos[index];
         setTodos(todos.filter((v,idx) => idx!==index));
+        fetch(`api/todos/${todoToRemove}`, {method: "delete"}).then((response) => response.json());
     }
 
+    function markAsDone( { index }) {
+        const completedToDo = todos[index]
+        fetch(`api/todos/${completedToDo.id}`, {method: "put", body: JSON.stringify({value: [...todos, index], 
+            done: !completedToDo.done})}).then((response) => response.json())
+            
+        setTodos ((oldTodos) => {
```

Comment:
set todos should take an array and pretty sure this is not being called after fetch?